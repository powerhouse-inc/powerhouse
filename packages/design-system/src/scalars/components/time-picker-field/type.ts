export type TimeFieldValue = string | null;
