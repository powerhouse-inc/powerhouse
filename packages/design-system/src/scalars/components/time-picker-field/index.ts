export { TimePickerField, type TimeFieldProps } from "./time-picker-field.js";
