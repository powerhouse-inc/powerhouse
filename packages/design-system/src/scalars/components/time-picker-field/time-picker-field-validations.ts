import { type TimeFieldProps } from "#scalars";
import {
  convert24hTo12h,
  getHoursAndMinutesFromValue,
  getTime,
  isFormatTimeAllowed,
  isValidTimeFromValue,
  TIME_PATTERNS,
} from "../../../ui/components/data-entry/time-picker/utils.js";

export const validateTimePicker =
  ({ timeFormat }: TimeFieldProps) =>
  (value: unknown) => {
    if (value === "") {
      return true;
    }
    if (value === undefined) {
      return true;
    }
    const getHoursAndMinutes = getHoursAndMinutesFromValue(value as string);

    if (!isValidTimeFromValue(getHoursAndMinutes)) {
      return "Invalid time. Please enter time a valid format";
    }
    // Validate the format of the value
    const time = getTime((value as string).trim());
    if (!timeFormat) {
      const is24HourValid = TIME_PATTERNS.HOURS_24.test(time);
      const is12HourValid = TIME_PATTERNS.HOURS_12.test(time);
      if (!is24HourValid && !is12HourValid) {
        return "Invalid time. Please enter time a valid format";
      }
    }
    if (timeFormat && !isFormatTimeAllowed(timeFormat)) {
      return `Invalid time format. Please select a valid format`;
    }
    // Validar formato 24-hour (HH:mm)
    if (timeFormat === "HH:mm") {
      if (!TIME_PATTERNS.HOURS_24.test(time)) {
        return "Please enter a valid time in 24-hour format";
      }
      // Additional validation for real hours/minutes
      const [hours, minutes] = time.split(":").map(Number);
      if (hours > 23 || hours < 0 || minutes > 59 || minutes < 0) {
        return "Please enter a valid time in 24-hour format";
      }
    }

    // Validar formato 12-hour (hh:mm a)
    if (timeFormat === "hh:mm a" || timeFormat === "hh:mm A") {
      const parse12HorsFormat = convert24hTo12h(time);

      // Additional validation for real hours/minutes
      const [timePart] = parse12HorsFormat.split(" ");
      const [hours, minutes] = timePart.split(":").map(Number);
      if (hours > 12 || hours < 1 || minutes > 59 || minutes < 0) {
        return "Please enter a valid time in 12-hour format";
      }
    }
    return true;
  };
