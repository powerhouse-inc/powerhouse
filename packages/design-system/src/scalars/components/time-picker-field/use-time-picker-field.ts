import React, { useMemo, useState } from "react";
import { TimeFieldValue, TimePeriod } from "./type";
import {
  cleanTime,
  getOffsetToDisplay,
  convert12hTo24h,
  createChangeEvent,
  formatInputToDisplayValid,
  formatInputsToValueFormat,
  getHours,
  getHoursAndMinutes,
  getInputValue,
  getMinutes,
  getOffset,
  getOptions,
  getTimezone,
  isValidTimeInput,
} from "./utils";
export const convertTimeFrom24To12Hours = (time: string) => {
  if (time === "") return "";
  const hours = Number(time);
  if (isNaN(hours)) return "";
  return String(hours % 12 || 12).padStart(2, "0");
};
interface TimePickerFieldProps {
  value?: TimeFieldValue;
  defaultValue?: TimeFieldValue;
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onBlur?: (e: React.FocusEvent<HTMLInputElement>) => void;
  timeFormat?: string;
  dateIntervals?: number;
  timeZone?: string;
  showTimezoneSelect?: boolean;
}

export const useTimePickerField = ({
  value,
  defaultValue,
  onChange,
  onBlur,
  timeFormat = "hh:mm a",
  dateIntervals = 1,
  timeZone,
  showTimezoneSelect = true,
}: TimePickerFieldProps) => {
  const [isOpen, setIsOpen] = useState(false);
  const is12HourFormat = timeFormat.includes("a");
  const [inputValue, setInputValue] = useState(
    getInputValue(value ?? defaultValue),
  );

  const [selectedHour, setSelectedHour] = useState(
    is12HourFormat
      ? convertTimeFrom24To12Hours(getHours(value ?? defaultValue ?? ""))
      : getHours(value ?? defaultValue ?? ""),
  );

  const [selectedMinute, setSelectedMinute] = useState(
    getMinutes(value ?? defaultValue ?? ""),
  );

  const [selectedPeriod, setSelectedPeriod] = useState<TimePeriod | undefined>(
    undefined,
  );

  const [selectedTimeZone, setSelectedTimeZone] = useState<
    string | string[] | undefined
  >(getTimezone(value ?? defaultValue ?? ""));

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const input = e.target.value;
    setInputValue(input);
  };
  const handleBlur = (e: React.FocusEvent<HTMLInputElement>) => {
    const input = e.target.value;

    if (!isValidTimeInput(input)) {
      if (input === "") {
        setInputValue(input);
        onChange?.(createChangeEvent(""));
        onBlur?.(e);
        return;
      }
      const inValid = formatInputsToValueFormat("99", "99", "Â±09:09");
      setInputValue(input);
      onChange?.(createChangeEvent(inValid));
      onBlur?.(e);
      return;
    }
    const validDisplay = formatInputToDisplayValid(
      input,
      is12HourFormat,
      dateIntervals,
      selectedPeriod,
    );
    setInputValue(validDisplay);

    const validValue = convert12hTo24h(validDisplay);

    const { minutes, hours, period } = getHoursAndMinutes(validValue);

    const offsetUTC = getOffset(selectedTimeZone as string);

    if (is12HourFormat) {
      setSelectedPeriod(period as TimePeriod);
    }

    const datetime = formatInputsToValueFormat(hours, minutes, offsetUTC);
    const clearMinutes = cleanTime(minutes);
    const clearHours = convertTimeFrom24To12Hours(cleanTime(hours));
    setSelectedHour(clearHours);
    setSelectedMinute(clearMinutes);
    onChange?.(createChangeEvent(datetime));
    onBlur?.(e);
  };

  // Generate hours according to the format
  const hours = is12HourFormat
    ? Array.from({ length: 12 }, (_, i) => String(i + 1).padStart(2, "0")) // 1-12
    : Array.from({ length: 24 }, (_, i) => String(i).padStart(2, "0")); // 0-23

  const minutes = useMemo(() => {
    if (dateIntervals > 1) {
      const arr: string[] = [];
      for (let i = 0; i < 60; i += dateIntervals) {
        arr.push(String(i).padStart(2, "0"));
      }
      return arr;
    }
    return Array.from({ length: 60 }, (_, i) => String(i).padStart(2, "0"));
  }, [dateIntervals]);

  const handleSave = () => {
    setIsOpen(false);
    const offsetUTC = getOffset(selectedTimeZone as string);

    // Value to save in the onSubmin
    const datetime = formatInputsToValueFormat(
      selectedHour,
      selectedMinute,
      offsetUTC,
    );
    // Value to display in the input
    const validDisplay = formatInputToDisplayValid(
      datetime,
      is12HourFormat,
      dateIntervals,
      selectedPeriod,
    );
    setInputValue(validDisplay);
    onChange?.(createChangeEvent(datetime));
  };

  const handleCancel = () => {
    setIsOpen(false);
  };

  const options = useMemo(() => {
    return getOptions();
  }, []);

  // if timeZone, then the options of select will be that timeZone and the offset
  const isDisableSelect = timeZone || !showTimezoneSelect ? true : false;
  const timeZonesOptions = timeZone
    ? [
        options.find((opt) => opt.value === timeZone) || {
          label: `(${getOffsetToDisplay(timeZone)}) ${timeZone.replace(/_/g, " ")}`,
          value: timeZone,
        },
      ]
    : options;

  return {
    selectedHour,
    selectedMinute,
    selectedPeriod,
    setSelectedHour,
    setSelectedMinute,
    setSelectedPeriod,
    hours,
    minutes,
    inputValue,
    handleInputChange,
    isOpen,
    setIsOpen,
    handleCancel,
    handleSave,
    timeZonesOptions,
    handleBlur,
    timeFormat,
    is12HourFormat,
    selectedTimeZone,
    setSelectedTimeZone,
    isDisableSelect,
  };
};
