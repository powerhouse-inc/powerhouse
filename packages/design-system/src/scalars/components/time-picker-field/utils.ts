import { TimePeriod } from "./type";

export const createChangeEvent = (
  value: string,
): React.ChangeEvent<HTMLInputElement> => {
  const nativeEvent = new Event("change", {
    bubbles: true,
    cancelable: true,
  });

  Object.defineProperty(nativeEvent, "target", {
    value: { value },
    writable: false,
  });

  return nativeEvent as unknown as React.ChangeEvent<HTMLInputElement>;
};

export const ALLOWED_TIME_FORMATS = ["hh:mm a", "HH:mm", "hh:mm A"];

export const isFormatTimeAllowed = (format: string): boolean => {
  return ALLOWED_TIME_FORMATS.includes(format);
};

export const TIME_PATTERNS = {
  /** Matches times in 24-hour format (HH:mm) like: 0:00, 09:30, 23:59 */
  HOURS_24: /^([0-9]|0[0-9]|1[0-9]|2[0-3]):([0-5][0-9])$/,
  /** Matches times in 12-hour format (h:mm a) like: 1:00 AM, 12:59 PM */
  HOURS_12: /^(0?[1-9]|1[0-2]):([0-5][0-9])\s(AM|PM|am|pm)$/,
} as const;

export const roundMinute = (minute: number, interval: number): number => {
  const remainder = minute % interval;
  let roundedMinute: number;
  if (remainder >= interval / 2) {
    roundedMinute = minute + (interval - remainder);
    if (roundedMinute >= 60) {
      // If it exceeds 60, round down
      roundedMinute = minute - remainder;
    }
  } else {
    roundedMinute = minute - remainder;
  }
  return roundedMinute;
};

export const transformInputTime = (
  input: string,
  is12HourFormat: boolean,
  defaultPeriod?: TimePeriod,
): { hour: string; minute: string; period?: TimePeriod } => {
  input = input.trim();
  let hourStr = "";
  let minuteStr = "";
  let period: TimePeriod | undefined = undefined;

  if (input.includes(":")) {
    const [hourPart, rest] = input.split(":", 2);
    hourStr = hourPart;
    const parts = rest.split(/\s+/);
    minuteStr = parts[0];
    if (parts.length > 1) {
      period = parts[1].toUpperCase() as TimePeriod;
    }
  } else {
    // Handle short hours: 8 -> 08:00, 12 -> 12:00
    const digits = input.padStart(4, "0");
    hourStr = digits.slice(0, 2);
    minuteStr = digits.slice(2, 4);
  }

  let hourNum = parseInt(hourStr, 10);
  let minuteNum = parseInt(minuteStr, 10) || 0;
  if (isNaN(hourNum)) hourNum = 0;
  if (isNaN(minuteNum)) minuteNum = 0;

  if (is12HourFormat) {
    if (!period) {
      period = defaultPeriod || "AM";
      if (hourNum === 12) period = "PM";
    }

    // Convert 24h format to 12h
    if (hourNum > 12) {
      period = "AM";
      hourNum -= 12;
    }

    // Special case: 0 -> 12 AM
    if (hourNum === 0) {
      hourNum = 12;
    }

    // Special rule: 12 without period -> PM
    period = hourNum >= 8 && hourNum <= 11 ? "AM" : "PM";
  } else {
    {
      // Convert any AM/PM designator to 24h
      if (period === "PM" && hourNum < 12) {
        hourNum += 12;
      } else if (period === "AM" && hourNum === 12) {
        hourNum = 0;
      }
      // Force period to undefined in 24h format
      period = undefined;
    }
  }

  return {
    hour: String(hourNum).padStart(2, "0"),
    minute: String(minuteNum).padStart(2, "0"),
    period: is12HourFormat ? period : undefined,
  };
};

export const isValidTimeInput = (input: string): boolean => {
  input = input.trim();
  // Allow change the format and convert the time
  if (input.includes(":")) {
    return (
      TIME_PATTERNS.HOURS_12.test(input) || TIME_PATTERNS.HOURS_24.test(input)
    );
  } else {
    // "For digits, expect 3 or 4 characters and validate they're in range".
    if (input.length !== 3 && input.length !== 4) {
      return false;
    }
    const hourStr = input.length === 3 ? input.slice(0, 1) : input.slice(0, 2);
    const minuteStr = input.slice(-2);
    const hourNum = parseInt(hourStr, 10);
    const minuteNum = parseInt(minuteStr, 10);
    return hourNum >= 0 && hourNum <= 23 && minuteNum >= 0 && minuteNum <= 59;
  }
};
