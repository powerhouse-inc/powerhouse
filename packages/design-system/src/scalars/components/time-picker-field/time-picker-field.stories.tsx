import type { Meta, StoryObj } from "@storybook/react";
import { withForm } from "../../lib/decorators.js";
import {
  getDefaultArgTypes,
  getValidationArgTypes,
  StorybookControlCategory,
} from "../../lib/storybook-arg-types.js";
import { TimePickerField } from "./time-picker-field.js";

const meta: Meta<typeof TimePickerField> = {
  title: "Document Engineering/Scalars/TimePickerField",
  component: TimePickerField,
  parameters: {
    layout: "centered",
    form: {
      resetBehavior: "unmount",
    },
  },
  decorators: [withForm],
  tags: ["autodocs"],
  argTypes: {
    ...getDefaultArgTypes(),
    ...getValidationArgTypes(),
    timeFormat: {
      control: {
        type: "select",
      },
      description: "The format of the time in the time picker",
      table: {
        type: { summary: "string" },
        category: StorybookControlCategory.COMPONENT_SPECIFIC,
      },

      options: ["hh:mm A", "HH:mm"],
      defaultValue: { summary: "hh:mm A" },
    },
    showTimezoneSelect: {
      control: {
        type: "boolean",
      },
      description: "Show timezone select",
      table: {
        category: StorybookControlCategory.COMPONENT_SPECIFIC,
        type: { summary: "boolean" },
      },
    },
    timeIntervals: {
      description: "The interval between each time option",
      control: {
        type: "number",
      },

      type: "number",
      min: 1,
      max: 60,
      table: {
        category: StorybookControlCategory.COMPONENT_SPECIFIC,
      },

      defaultValue: { summary: 1 },
    },
    timeZone: {
      description: "The timezone to display in the time picker",
      control: {
        type: "text",
      },
      table: {
        category: StorybookControlCategory.COMPONENT_SPECIFIC,
        type: { summary: "string" },
      },
    },
    includeContinent: {
      description: "Show continent name in the timezone select",
      control: {
        type: "boolean",
        defaultValue: false,
      },
      table: {
        category: StorybookControlCategory.COMPONENT_SPECIFIC,
        defaultValue: { summary: "false " },
        type: { summary: "boolean" },
      },
    },
    placeholder: {
      description: "The placeholder text for the time picker",
      table: {
        category: StorybookControlCategory.DEFAULT,
        type: { summary: "string" },
      },
    },
  },

  args: {
    name: "time-picker-field",
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    name: "time",
    label: "Pick a time",
    placeholder: "HH:mm",
  },
};
export const Disabled: Story = {
  args: {
    name: "time",
    label: "Pick a time",
    placeholder: "HH:mm",
    disabled: true,
  },
};

export const Filled: Story = {
  args: {
    name: "time",
    label: "Pick a time",
    value: "12:00 PM",
    placeholder: "HH:mm",
  },
};
