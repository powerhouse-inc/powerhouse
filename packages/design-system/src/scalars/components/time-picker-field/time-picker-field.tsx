import {
  TimePicker,
  type TimePickerProps,
} from "../../../ui/components/data-entry/time-picker/time-picker.js";
import { withFieldValidation } from "../fragments/with-field-validation/with-field-validation.js";
import { type FieldErrorHandling } from "../types.js";
import { validateTimePicker } from "./time-picker-field-validations.js";

interface TimeFieldProps extends TimePickerProps, FieldErrorHandling {}

const TimePickerField = withFieldValidation<TimeFieldProps>(TimePicker, {
  validations: {
    _timePickerType: validateTimePicker,
  },
});

TimePickerField.displayName = "TimePickerField";

export { TimePickerField, type TimeFieldProps };
