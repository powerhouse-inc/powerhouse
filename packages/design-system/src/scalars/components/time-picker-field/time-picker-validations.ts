import { TimePickerFieldProps } from "./time-picker-field";
import { isFormatTimeAllowed } from "./utils";

export const validateTimePicker =
  ({ timeFormat }: TimePickerFieldProps) =>
  (value: unknown) => {
    if (value === "") {
      return true;
    }
    if (value === undefined) {
      return true;
    }

    if (timeFormat && !isFormatTimeAllowed(timeFormat)) {
      return `Invalid time format.Plese select a valid format`;
    }

    return true;
  };
