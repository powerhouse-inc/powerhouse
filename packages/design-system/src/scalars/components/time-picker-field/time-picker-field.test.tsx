import { screen } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import { vi } from "vitest";
import { renderWithForm } from "../../lib/testing.js";

import { TimePickerField } from "./time-picker-field.js";
vi.mock("#powerhouse", () => ({
  Icon: ({ name, className }: { name: string; className?: string }) => (
    <div data-testid={`mock-icon-${name}`} className={className}>
      Mock Icon: {name}
    </div>
  ),
}));

describe("TimePickerField", () => {
  it("should match the snapshot", () => {
    const { container } = renderWithForm(
      <TimePickerField
        label="Test Label"
        name="test-time"
        id="test-id"
        required
        disabled={false}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it("should display the label when provided", () => {
    const labelText = "Test Label";
    renderWithForm(<TimePickerField name="test-time" label={labelText} />);
    expect(screen.getByText(labelText)).toBeInTheDocument();
  });

  it("should not render the label when label prop is not provided", () => {
    renderWithForm(<TimePickerField name="test-time" />);
    expect(screen.queryByText("Test Label")).not.toBeInTheDocument();
  });

  it("should mark the label as required when required prop is true", () => {
    renderWithForm(
      <TimePickerField name="test-time" label="Test Label" required />,
    );
    const label = screen.getByText("Test Label");
    const asterisk = screen.getByText("*");
    expect(label).toBeInTheDocument();
    expect(asterisk).toBeInTheDocument();
  });

  it("should mark the label as disabled when disabled prop is true", () => {
    renderWithForm(
      <TimePickerField name="test-time" label="Test Label" disabled />,
    );
    const label = screen.getByText("Test Label");
    expect(label).toHaveClass("cursor-not-allowed");
    expect(label).toHaveClass("text-gray-700");
  });

  it("should handle timeIntervals prop correctly with 15-minute intervals", async () => {
    const user = userEvent.setup();
    renderWithForm(
      <TimePickerField
        name="test-time"
        label="Test Label"
        timeIntervals={15}
      />,
    );
    // Open the time picker using the button instead of the input
    const clockButton = screen.getByRole("button");
    await user.click(clockButton);

    // First wait for the popover to be visible
    const popoverContent = await screen.findByRole("dialog");
    expect(popoverContent).toBeVisible();

    // Now check for the minute values
    const minutes = ["00", "15", "30", "45"];
    for (const minute of minutes) {
      // Use findAllByText instead of findByText since there are multiple elements with the same text
      const minuteElements = await screen.findAllByText(minute);
      expect(minuteElements.length).toBeGreaterThan(0);
      expect(minuteElements[0]).toBeInTheDocument();
    }

    // Verify that intermediate values are not present
    ["13", "20", "25", "35", "40", "50", "55"].forEach((minute) => {
      expect(screen.queryByText(minute)).not.toBeInTheDocument();
    });
  });

  it("should handle timeIntervals prop correctly with 30-minute intervals", async () => {
    const user = userEvent.setup();
    renderWithForm(
      <TimePickerField
        name="test-time"
        label="Test Label"
        timeIntervals={30}
      />,
    );

    // Open the time picker using the button instead of the input
    const clockButton = screen.getByRole("button");
    await user.click(clockButton);

    // First wait for the popover to be visible
    const popoverContent = await screen.findByRole("dialog");
    expect(popoverContent).toBeVisible();

    // Now check for the minute values
    const minutes = ["00", "30"];
    for (const minute of minutes) {
      const minuteElements = await screen.findAllByText(minute);
      expect(minuteElements.length).toBeGreaterThan(0);
      expect(minuteElements[0]).toBeInTheDocument();
    }
    // Verify that other values are not present
    ["15", "45"].forEach((minute) => {
      expect(screen.queryByText(minute)).not.toBeInTheDocument();
    });
  });

  it("should set timezone value and disable select when timeZone prop is provided", async () => {
    const user = userEvent.setup();
    const timeZoneValue = "America/New_York";

    renderWithForm(
      <TimePickerField
        name="test-time"
        label="Test Label"
        timeZone={timeZoneValue}
      />,
    );

    const clockButton = screen.getByRole("button");
    await user.click(clockButton);

    const popoverContent = await screen.findByRole("dialog");
    expect(popoverContent).toBeVisible();

    const select = await screen.findByRole("combobox");
    expect(select).toBeDisabled();

    // Use a more flexible approach to find the timezone text
    const timezoneText = screen.getByText((content) => {
      return content.includes("New York") && content.includes("(");
    });
    expect(timezoneText).toBeInTheDocument();
  });

  it("should display continent when includeContinent prop is true", async () => {
    const user = userEvent.setup();
    const timeZoneValue = "America/New_York";

    renderWithForm(
      <TimePickerField
        name="test-time"
        label="Test Label"
        timeZone={timeZoneValue}
        includeContinent
      />,
    );

    const clockButton = screen.getByRole("button");
    await user.click(clockButton);

    const popoverContent = await screen.findByRole("dialog");
    expect(popoverContent).toBeVisible();

    // Verify that the timezone includes the continent (America)
    const timezoneText = screen.getByText((content) => {
      return content.includes("America") && content.includes("New York");
    });
    expect(timezoneText).toBeInTheDocument();
  });

  it("should not display continent when includeContinent prop is false", async () => {
    const user = userEvent.setup();
    const timeZoneValue = "America/New_York";

    renderWithForm(
      <TimePickerField
        name="test-time"
        label="Test Label"
        timeZone={timeZoneValue}
        includeContinent={false}
      />,
    );

    const clockButton = screen.getByRole("button");
    await user.click(clockButton);

    const popoverContent = await screen.findByRole("dialog");
    expect(popoverContent).toBeVisible();

    // Verify that the timezone excludes the continent (America)
    const timezoneText = screen.getByText((content) => {
      return content.includes("New York") && !content.includes("America");
    });
    expect(timezoneText).toBeInTheDocument();
  });
});
