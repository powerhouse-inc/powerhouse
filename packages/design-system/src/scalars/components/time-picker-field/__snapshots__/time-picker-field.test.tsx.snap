// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TimePickerField > should match the snapshot 1`] = `
<div>
  <form
    id=":r0:"
    novalidate=""
  >
    <div
      class="flex flex-col gap-2"
    >
      <label
        class="inline-flex items-center text-sm font-medium leading-4 text-gray-900 dark:text-gray-50 mb-[3px]"
        for="test-id"
      >
        Test Label
        <span
          class="ml-1 text-gray-800 group-hover:text-gray-900 dark:text-gray-400 dark:group-hover:text-slate-50"
        >
          *
        </span>
      </label>
      <div
        class="flex flex-col space-y-2"
      >
        <div
          class="flex w-[275px] rounded-md text-sm focus-within:ring-ring focus-within:ring-1 focus-within:ring-offset-0 ring-gray-900 dark:ring-charcoal-300 dark:border-charcoal-700 dark:bg-charcoal-900 border border-gray-300 bg-white hover:border-gray-300 hover:bg-gray-100 dark:hover:bg-charcoal-800 dark:hover:border-charcoal-700 [&:hover_.input-field]:bg-transparent [&:hover_.button-ghost]:bg-transparent focus:[&_.input-field]:bg-transparent focus-within:hover:bg-transparent focus-within:hover:cursor-default"
        >
          <button
            aria-controls="radix-:r1:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:pointer-events-none [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent hover:text-accent-foreground w-[50px] rounded-l-md border-none px-2 focus:bg-none focus:text-selected-foreground button-ghost"
            data-state="closed"
            type="button"
          >
            <div
              class="size-4 hover:none text-gray-700 dark:text-gray-50"
              data-testid="mock-icon-Clock"
            >
              Mock Icon: 
              Clock
            </div>
          </button>
          <input
            class="flex h-9 rounded-md text-sm font-normal leading-5 text-gray-900 dark:text-gray-50 dark:border-charcoal-700 dark:bg-charcoal-900 border border-gray-300 bg-white px-3 py-2 font-sans placeholder:text-gray-500 dark:placeholder:text-gray-600 focus-visible:outline-none focus-visible:ring-gray-900 focus-visible:ring-offset-white dark:focus-visible:ring-charcoal-300 dark:focus-visible:ring-offset-charcoal-900 dark:focus:bg-charcoal-900 disabled:cursor-not-allowed disabled:border-gray-300 disabled:bg-white disabled:text-gray-700 disabled:dark:border-charcoal-800 disabled:dark:bg-charcoal-900 disabled:dark:text-gray-300 w-full rounded-l-none border-none text-right placeholder:text-right focus-visible:ring-0 focus-visible:ring-offset-0 focus:bg-white input-field"
            name="test-time"
            required=""
            value=""
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;
